graphstorm-processing/graphstorm_processing/config/numerical_configs.py

```diff
@@ -92,3 +92,53 @@ def __init__(self, config: Mapping):
         self.separator = self._transformation_kwargs.get("separator", None)
 
         self._sanity_check()
+
+
+class BucketFeatureConfig(FeatureConfig):
+    """Feature configuration for bucket normalization features.
+
+    Supported kwargs
+    ----------------
+    imputer: str
+        A method to fill in missing values in the data. Valid values are:
+        "none" (Default), "mean", "median", and "most_frequent". Missing values will be replaced
+        with the respective value computed from the data.
+
+    bucket_cnt: int
+        The count of bucket lists used in the bucket feature transform. Each bucket will
+        have same length.
+
+    range: List[float]
+        The range of bucket lists only defining the start and end point. The range and
+        bucket_cnt will define the buckets together. For example, range = [10, 30] and
+        bucket_cnt = 2, will have two buckets: [10, 20] and [20, 30]
+
+    slide_window_size: float or none
+        Interval or range within which numeric values are grouped into buckets. Slide window
+        size will let one value possibly fall into multiple buckets.
+    """
+
+    def __init__(self, config: Mapping):
+        super().__init__(config)
+        self.imputer = self._transformation_kwargs.get("imputer", "none")
+        self.bucket_cnt = self._transformation_kwargs.get("bucket_cnt", "none")
+        self.range = self._transformation_kwargs.get("range", "none")
+        self.slide_window_size = self._transformation_kwargs.get("slide_window_size", "none")
+        self._sanity_check()
+
+    def _sanity_check(self) -> None:
+        super()._sanity_check()
+        assert (
+            self.imputer in VALID_IMPUTERS
+        ), f"Unknown imputer requested, expected one of {VALID_IMPUTERS}, got {self.imputer}"
+        assert isinstance(
+            self.bucket_cnt, int
+        ), f"Expect bucket_cnt {self.bucket_cnt} be an integer"
+        assert (
+            isinstance(self.range, list)
+            and all(isinstance(x, (int, float)) for x in self.range)
```

Comment:
Better to use `numbers.Number` here see: https://stackoverflow.com/questions/4187185/how-can-i-check-if-my-python-object-is-a-number